LSP-lemminx.sublime-settings

```diff
@@ -1,4 +1,5 @@
 {
+	"command": ["java", "-jar", "${cache_path}/LSP-lemminx/org.eclipse.lemminx-${version}-uber.jar"],
```

Comment:
IMO command should not be exposed. One of the points of those packages is that such things are hidden away and just work.

Reply:
I disagree. This way a user can use a plugin package as well as experiment around with their own binaries.

Reply:
It also confuses normal users as then they think they have to actually change it to their own path.

We recently had people on the channel that thought they have to install their own dependencies even when using lsp-intelephense.

Reply:
I already struggled with a machine on which java was not added to PATH for security reasons. Adding `JAVA_HOME` to the `"env"` dict and extending PATH didn't work. Need a way to handle such situations.

Reply:
You can still set a custom command from within code as follows:

```python

class Lemminx(AbstractPlugin):

    @classmethod
    def name(cls) -> str:
        return "lemminx"

    @classmethod
    def configuration(cls) -> sublime.Settings:
        settings = sublime.load_settings("LSP-{}.sublime-settings".format(cls.name())
        setting.set("command", ...)
        return settings

```

Reply:
> We recently had people on the channel that thought they have to install their own dependencies even when using lsp-intelephense.

From what I understood that user got confused becauase there's still an intelephense default config in LSP.sublime-settings.

Reply:
If the user customizes it then he loses the benefit of automatic updates.
If the user wants to have custom command then maybe he shouldn't be using a plugin in the first place? (That means he wouldn't get the benefit of custom handlers/notification if there are some, though).

In any case, I wouldn't want it for the single reason that I think it would confuse people. The see the path and they think it's incorrect and need to set their own. So lets at least try to address that concern while still allowing customization. Maybe:
 - allow customizing but have it in the default settings file
 - have it in settings file but commented out and with a comment explaining the purpose of it.

Reply:
In general I am with @rchl not to expose command details which are not neccessary. In case of language servers running in a VM (by node, python, java, ...) a possibility to define the interpreter path should be given. I would even be fine with extending the `PATH` via "env" would work. Haven't had time to track down why it does not though.

Reply:
OK, I'll adjust this draft PR.